### PR TITLE
Added support for GET '/datasets/{id}' Issue #523

### PR DESCRIPTION
**Pull request by pcingola**
Added support for GET '/datasets/{id}' Issue #523

### CODE REVIEW

File: ga4gh/cli.py [hunk 1]
```diff
@@ -936,6 +956,7 @@ def client_main(parser=None):
     addCallsetsGetParser(subparsers)
     addVariantsGetParser(subparsers)
     addReferencesBasesListParser(subparsers)
+    addDatasetsGetParser(subparsers)
```
Comment (dcolligan) [inline ga4gh/cli.py]: switch this line with the line above, so we have all the add*GetParser calls together, just for clarity

**Comment by dcolligan** [comment]
@pcingola this is a good start.  The tests are failing, however.  Run `python script/run_tests.py` to run the Travis tests locally.  You should also add a test that tests that the route you added works... see the other tests in the same file in which the current failing test is located for examples.

**Comment by pcingola** [comment]
OK, apparently installing "virtualenv" is a must for the tests to pass. I somehow I missed it in the documentation (it appeared to be optional).
Also, a module called 'funcsigs' is also necessary but it is either not in the requirements.txt or for some reason is missing on my computer....
After that, the tests are passing.

How can I update the documentation? There are several examples that simply do not work, which is a complication when you are trying to learn how things work...(or even how to install a development environment).

**Comment by pcingola** [comment]
Added test case 'tests.unit.test_views.TestFrontend.testGetDataset'

**Comment by dcolligan** [comment]
@pcingola yes, the documentation is out of date.  There is an issue to remedy that: #595

I doubt there is a module required for the project that is not in `requirements.txt` (or the packages that the packages in `requirements.txt` depend on, which are automatically installed with a normal installation), otherwise the tests would fail.

The docs are in the `docs` subdirectory; they are tracked just like any other file in our project.  We use sphinx for our docs.  `run_tests` will build the docs as part of its run, but you can do the same manually with:

```
make clean -C docs/source/ && make -C docs/source/
```

File: tests/unit/test_views.py [hunk 2]
```diff
@@ -133,6 +133,13 @@ def sendGetVariantSet(self, id_=None):
         response = self.sendGetRequest(path)
         return response
 
+    def sendGetDataset(self, id_=None):
+        if id_ is None:
+            id_ = self.variantSetId
```
Comment (dcolligan) [inline tests/unit/test_views.py]: why is this `self.variantSetId`?  shouldn't it be a dataset id?

File: tests/unit/test_views.py [hunk 3]
```diff
@@ -358,7 +373,6 @@ def testNotImplementedPaths(self):
         pathsNotImplementedPost = [
         ]
         pathsNotImplementedGet = [
-            '/datasets/<id>',
```
Comment (dcolligan) [inline tests/unit/test_views.py]: we can just delete this whole test now, since it doesn't do anything anymore

**Comment by dcolligan** [comment]
This is progressing nicely, but I would like to see two more tests, one for the client and one for the cli.  Do something in `tests/unit/test_cli.py` and `tests/unit/test_client.py` analogous to what I did in 2f6180120db855109b9baff24e132b5531d70a8c for the get callset endpoint.

File: tests/unit/test_views.py [hunk 4]
```diff
@@ -290,6 +297,14 @@ def testVariantSetsSearch(self):
             response.data)
         self.assertEqual(len(responseData.variantSets), 1)
 
+    def testGetDataset(self):
+        response = self.sendDatasetsSearch()
+        responseData = protocol.SearchDatasetsResponse.fromJsonString(
+            response.data)
+        datasetId = responseData.datasets[0].id
+        response = self.sendGetDataset(datasetId)
+        self.assertEqual(200, response.status_code)
```
Comment (jeromekelleher) [inline tests/unit/test_views.py]: We need to test the 'not found' condition here too, which should give us a 404 HTTP response. See the second half of testGetVariantSet for a model to follow.

**Comment by jeromekelleher** [comment]
This looks great @pcingola. I agree with the comments @dcolligan makes above.

I'd also like to see two more tests:

1. Add a line to ``testCors`` in test_views.py.
2. Add a ``testGetDataset()`` in tests/unit/test_simulated_stack.py. Use ``testGetReferenceSet`` as a model.

This is quite a few different tests, but we're touching a lot of different parts of the codebase here, on both the client and server side.

**Comment by pcingola** [comment]
Added your suggestions and tests.
Thank you for your feedback!

**Comment by jeromekelleher** [comment]
Excellent, thanks @pcingola! A few more very minor things and I'm ready to +1 and merge:

1. We still need the 404 check in testGetDataset in tests_views. You can pretty much copy the second half of testGetVariantSet.
2 The commented out code for ``testNotImplementedPaths`` should be deleted.
3. Once that's done, the commits need to be squashed down to one commit, and rebased on upstream/develop. I think there's a section on this in the develop docs.

**Comment by pcingola** [comment]
OK, added the 404 check and deleted testNotImplementedPaths.

Created a new PR #611 with all commits squashed.

**Comment by dcolligan** [comment]
Closing this PR since there is a newer one open.

Btw @pcingola usually the best practice is to keep the same PR open and force push to the same branch that you are working on if you have modified the commits.  That way the comments are not split across two PRs for the same change.

**Pull request closed without merging**